master/src/main/scala/org/apache/celeborn/service/deploy/master/Master.scala

```diff
@@ -850,6 +852,22 @@ private[celeborn] class Master(
       logInfo(s"Offered extra $offerSlotsExtraSize slots for $shuffleKey")
     }
 
+    if (authEnabled) {
+      // Pass application registration information to the workers
+      val pbApplicationMeta = PbApplicationMeta.newBuilder()
+        .setAppId(requestSlots.applicationId)
+        .setSecret(secretRegistry.getSecretKey(requestSlots.applicationId))
+        .build()
+      val transportMessage =
+        new TransportMessage(MessageType.APPLICATION_META, pbApplicationMeta.toByteArray)
+
+      slots.keySet().asScala.foreach { worker =>
+        logDebug(s"Sending app registration info to ${worker.host}:${worker.internalPort}")
+        internalRpcEnvInUse.setupEndpointRef(
```

Comment:
IMO there may be a large number of workers needing to connect for the first time, and to avoid blocking the requestSlots requests, maybe better move the setupEndpoint  process to somewhere, such as when the worker register. WDYT?

Reply:
I can delegate this to a threadpool so it doesn't block the requestSlots request. Pushing `applicationMeta` is anyways best effort.

Reply:
Made the change. PTAL.